README.md

```diff
@@ -2,31 +2,365 @@
+    this.name = name;
+    this.breed = breed;
+    this.age = age;
+    this.waggingTail = waggingTail;
+}
+
+// Constructor #2
+public Dog(String name, int age) {
+    this.name = name;
+    this.age = age;
+    this.breed = "unknown";
+}
+```
+
+We'll update the `main` to create a third dog using the new constructor:
+
+```java
+public static void main(String[] args) {
+    Dog bigDog = new Dog("Fido", "Great Dane", 8, true);             // Constructor #1
+    Dog smallDog = new Dog("Sweetie", "Miniature Poodle", 2, false); // Constructor #1
+    Dog mediumDog = new Dog("Honey", 10);                            // Constructor #2
+}
+```
+
+![3 dogs](https://curriculum-content.s3.amazonaws.com/6676/java-methods/constructor_3dogs.png)
+
+There are 2 important things to note when defining constructors:
+
+1. Not all fields can have default values - for example, what could a "default"
+   name be for a dog? 
+2. Some fields do make sense to have default value - for example, we often don't know
+   the breed, so the second constructor defaults the `breed` field to "unknown".
+
+
+
+## Comprehension Check
+
+<details>
+    <summary>What is printed when the <code>main</code> method executes?</summary>
+
+  <p>Whiskers,true,squeaky mouse</p>
+  <p>Constructor #1 is executed since only two arguments are passed into the method.</p>
+
+</details>
+
+<br>
+
+```java
+public class Cat {
+    private String name;
+    private boolean purring;
+    private String favoriteToy;
+
+    // Constructor #1
+    public Cat(String name, boolean purring) {
+        this.name = name;
+        this.purring = purring;
+        this.favoriteToy = "squeaky mouse";
+    }
+
+    // Constructor #2
+    public Cat(String name, boolean purring, String favoriteToy) {
+        this.name = name;
+        this.purring = purring;
+        this.favoriteToy = favoriteToy;
+    }
+
+    public static void main(String[] args) {
+        Cat kitty = new Cat("Whiskers", true);
+        System.out.println(kitty.name + "," + kitty.purring + "," + kitty.favoriteToy);
+    }
+}
+```
+
+## Constructor Chaining
+
+We have some code duplication between the two constructors, specifically the
+two statements that assign `name` and `age`.  We can do something
+called *constructor chaining*, where one constructor calls another to initialize
+some variables.
+
+We can have the constructor that takes four parameters call the
+constructor that only takes two parameters to do the `name` and `age`
+initialization.  This must be done as the first line of code in the
+constructor body and the method name `this` is used to call the
+constructor, i.e.  `this(name, age);`.
+
+```java
+// Constructor #1
+public Dog(String name, int age) {
+    this.name = name;
+    this.age = age;
+    this.breed = "unknown";
+}
+
+// Constructor #2
+public Dog(String name, String breed, int age, boolean waggingTail) {
+    this(name, age);  // calls Constructor #1
```

Comment:
Consider having constructor 1 call constructor 2
```
public Dog(String name, int age) {
 this(name, "Unknown", age, false);
}
```

Reply:
yes, that is much better!